### DISCUSSION TITLE
How can i make a whole image as a button?

### DISCUSSION THREAD

**Discussion by pgraca97** [TomSchimansky/CustomTkinter#906] [General]
I have looked everywhere and tried everything, but it seems that custom tkinter doesn't let me make a whole image as a button. Like i don't want that my image is turned into a tiny icon on the button. Help???

![image](https://user-images.githubusercontent.com/116584573/208506134-60497ff2-4a74-4fdc-89a7-f558eefff064.png)

**Comment by harshnagar** [comment]
You Need to remove the text in the Button by adding text="" in the widget:

Example:
```

from customtkinter import *
from PIL import Image

main = CTk()
main.geometry("200x200+500+500")
btn_img=CTkImage(dark_image=Image.open("Cached/delete_icon.png"), size=(30, 30))
del_btn = CTkButton(main,image=btn_img, text="", fg_color="red", corner_radius=20, width=10, height=10)
del_btn.place(x=50, y=50)
main.mainloop()

```
<img width="170" alt="Screenshot 2022-12-20 102105" src="https://user-images.githubusercontent.com/35253819/208586527-2c338faf-3404-4c3f-bd1e-c73b842aa189.png">

**Reply by pgraca97** [reply]
![image](https://user-images.githubusercontent.com/116584573/208651609-90e62f1c-51d7-4a9c-83bb-768a36d3ddc9.png)
misson accomplished, thank you so much! 
But is there any way to make the button be "clicked down" as it happens on the standard tkinter library. Like when the button is clicked it is "pushed down" thanks!

**Comment by harshnagar** [comment]
Sorry, But I Think there is no Way to do so because CTk is Modern GUI and the thing you want is supported in older GUIs like Tk or TTK!

**Comment by Akascape** [comment]
>But is there any way to make the button be "clicked down" as it happens on the standard tkinter library. Like when the button is clicked it is "pushed down" thanks!

@pgraca97 There is one way to add a little animation in CTkButton:
**It will look like this:**

<img src="https://user-images.githubusercontent.com/89206401/208831254-0740110b-99d6-4128-822e-d143f79df3b4.gif" width=500, height=200>

### CODE
```python
import customtkinter
import threading

def run_animation():
    real_width = button.winfo_width()
    real_height = button.winfo_height()
    for i in range(1,4):
        button.configure(width=real_width-i, height=real_height-i)
    button.configure(width=real_width, height=real_height)
    
root = customtkinter.CTk()

button = customtkinter.CTkButton(root, command=lambda: threading.Thread(target=run_animation).start())
button.place(x=30, y=50)

root.mainloop()
```

**Reply by TomSchimansky** [reply]
This will not work on scaled displays as winfo_width returns the scaled width but with the configure method you set the width independent of scaling. So on a 2x scaled display this will blow up the button to double the size. Just to let you know...

**Reply by TomSchimansky** [reply]
winfo_width and winfo_height will be implemented to work correct in the future, but currently they are not.

**Comment by RobbieKhan** [comment]
Wrote my own implementation of ImageButton widget. Attached a source code. Maybe someone will find it useful

![257f903beb4caa6afabfeba555b942a3](https://user-images.githubusercontent.com/53120051/214703880-806d7716-8190-45c7-afec-3a0debae90de.gif)

```
import customtkinter
import gc


class CTkImageButton(customtkinter.CTkLabel):
    shape = None

    def __init__(self, command=None, args=(), **kwargs):
        self.command = command
        self.args: tuple = args
        super().__init__(**kwargs)
        self._label.configure(borderwidth=1, cursor='hand2')
        self.press_cid = self.bind(sequence='<ButtonPress>', command=self.__on_press, add='+')
        self.release_cid = self.bind(sequence='<ButtonRelease>', command=self.__on_release, add='+')

    def __on_press(self, event):
        self.configure(relief='groove')

    def __on_release(self, event):
        self.configure(relief='flat')
        if self.command is not None:
            self.command(*self.args)

    def configure(self, require_redraw=False, **kwargs):
        if 'command' in kwargs:
            self.command = kwargs.pop('command')
        if 'command_args' in kwargs:
            self.args = kwargs.pop('command_args')
        if 'relief' in kwargs:
            self._label.configure(relief=kwargs.pop('relief'))
        if 'state' in kwargs:
            # Do not pop here!
            if kwargs.get('state').__eq__(customtkinter.DISABLED):
                self.unbind('<ButtonPress>', self.press_cid)
                self.unbind('<ButtonRelease>', self.release_cid)
            else:
                self.press_cid = self.bind(sequence='<ButtonPress>', command=self.__on_press, add='+')
                self.release_cid = self.bind(sequence='<ButtonRelease>', command=self.__on_release, add='+')
        super().configure(require_redraw=require_redraw, **kwargs)

    def destroy(self):
        self.unbind('<ButtonPress>', self.press_cid)
        self.unbind('<ButtonRelease>', self.release_cid)
        del self.press_cid
        del self.release_cid
        del self.command
        del self.args
        super().destroy()
        del self
        gc.collect(1)

```
